### PR TITLE
KOGITO-6130 Review containers used in tests

### PR DESCRIPTION
**Pull request by cristianonicolai**
- Disable DevServices from Quarkus where these are not used
- Review containers versions to avoid duplicates in the build pipeline

Assembly:
- https://github.com/kiegroup/kogito-runtimes/pull/1695
- https://github.com/kiegroup/kogito-apps/pull/1095
- https://github.com/kiegroup/kogito-examples/pull/940
- https://github.com/kiegroup/optaplanner/pull/1658
- https://github.com/kiegroup/optaplanner-quickstarts/pull/220

### CODE REVIEW

File: kogito-build/kogito-dependencies-bom/pom.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
   <properties>
     <!-- this version property is used in plugins but also in dependencies too -->
-    <version.io.quarkus>2.4.0.Final</version.io.quarkus>
+    <version.io.quarkus>2.4.1.Final</version.io.quarkus>
```
Comment (radtriste) [inline kogito-build/kogito-dependencies-bom/pom.xml:19]: if you are going to change that version, it needs to be done as well on opta* repositories and kogito-examples please :)

File: kogito-test-utils/src/main/java/org/kie/kogito/testcontainers/KogitoOracleSqlContainer.java [hunk 2]
```diff
@@ -35,7 +35,6 @@
     public static final String ORACLE_CONNECTION_URI = "kogito.persistence.oracle.connection.uri";
 
     public KogitoOracleSqlContainer() {
-        super("oracleinanutshell/oracle-xe-11g");
```
Comment (radtriste) [inline kogito-test-utils/src/main/java/org/kie/kogito/testcontainers/KogitoOracleSqlContainer.java:38]: why this removed ?

Comment (cristianonicolai) [reply kogito-test-utils/src/main/java/org/kie/kogito/testcontainers/KogitoOracleSqlContainer.java:38]: this image is quite old, looks unmaintained, so using the default provided via testcontaineirs seems a better option.

File: kogito-build/kogito-dependencies-bom/pom.xml (hunk 1 above)
Comment (cristianonicolai) [reply kogito-build/kogito-dependencies-bom/pom.xml:19]: sure

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
@tiagodolphine @cristianonicolai are we good to merge those ?

**Comment by radtriste** [comment]
do we want this in 1.14 ?

**Comment by cristianonicolai** [comment]
> do we want this in 1.14 ?

I think we need, to be able to upgrade Quarkus, I'll create the PRs. Im just waiting for related PRs to be green but its almost mission impossible :/

**Review by nmirasch — APPROVED** [review]

**Review by tiagodolphine — APPROVED** [review]

**Comment by cristianonicolai** [comment]
Jenkins rerun optaplanner-quickstarts tests

**Comment by cristianonicolai** [comment]
Jenkins rerun optaplanner-quickstarts tests

**Pull request merged**